src/services/k8s.py

```diff
@@ -27,6 +27,47 @@
 GROUP_VERSION_PARTS_COUNT = 2
 
 
+class K8sClientError(Exception):
```

Comment:
I don't think it's a good idea put this Error class under `services`, when you import the exception, you will break dependency flow if designed like this. For example, ideally we don't need to expose `services` layer to `routers` at all (even though the current structure already violates this, but we can refactor later), The router should only need to interact with the tools layer.

However, by placing `K8sClientError` inside services, we’re forcing the routers layer to import it and then cast it to HTTPException, which creates an unnecessary hard dependency from routers to services.

My suggestion is to move the custom exception class into `utils` package instead. A utility module is typically meant for cross-layer, shared objects, which fits this use case much better.